cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerDiagnosticsService.java

```diff
@@ -0,0 +1,108 @@
+import com.cloudera.api.swagger.model.ApiCollectDiagnosticDataArguments;
+import com.cloudera.api.swagger.model.ApiCommand;
+import com.sequenceiq.cloudbreak.client.HttpClientConfig;
+import com.sequenceiq.cloudbreak.cloud.scheduler.CancellationException;
+import com.sequenceiq.cloudbreak.cluster.api.ClusterDiagnosticsService;
+import com.sequenceiq.cloudbreak.cluster.service.ClusterClientInitException;
+import com.sequenceiq.cloudbreak.cm.client.ClouderaManagerApiClientProvider;
+import com.sequenceiq.cloudbreak.cm.client.ClouderaManagerClientInitException;
+import com.sequenceiq.cloudbreak.cm.client.retry.ClouderaManagerApiFactory;
+import com.sequenceiq.cloudbreak.cm.polling.ClouderaManagerPollingServiceProvider;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.cluster.Cluster;
+import com.sequenceiq.cloudbreak.polling.PollingResult;
+import com.sequenceiq.cloudbreak.service.CloudbreakException;
+import com.sequenceiq.common.model.diagnostics.CmDiagnosticsParameters;
+
+@Service
+@Scope("prototype")
+public class ClouderaManagerDiagnosticsService implements ClusterDiagnosticsService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClouderaManagerDiagnosticsService.class);
+
+    @Inject
+    private ClouderaManagerApiClientProvider clouderaManagerApiClientProvider;
+
+    @Inject
+    private ClouderaManagerApiFactory clouderaManagerApiFactory;
+
+    @Inject
+    private ClouderaManagerPollingServiceProvider clouderaManagerPollingServiceProvider;
+
+    private final Stack stack;
+
+    private final HttpClientConfig clientConfig;
+
+    private ApiClient client;
+
+    public ClouderaManagerDiagnosticsService(Stack stack, HttpClientConfig clientConfig) {
+        this.stack = stack;
+        this.clientConfig = clientConfig;
+    }
+
+    @PostConstruct
+    public void initApiClient() throws ClusterClientInitException {
+        Cluster cluster = stack.getCluster();
+        String cloudbreakAmbariUser = cluster.getCloudbreakAmbariUser();
+        String cloudbreakAmbariPassword = cluster.getCloudbreakAmbariPassword();
+        try {
+            client = clouderaManagerApiClientProvider
+                    .getClient(stack.getGatewayPort(), cloudbreakAmbariUser, cloudbreakAmbariPassword, clientConfig);
+        } catch (ClouderaManagerClientInitException e) {
+            throw new ClusterClientInitException(e);
+        }
+    }
+
+    @Override
+    public void collectDiagnostics(CmDiagnosticsParameters parameters) throws CloudbreakException {
+        ClouderaManagerResourceApi resourceApi = clouderaManagerApiFactory.getClouderaManagerResourceApi(client);
+        try {
+            ApiCommand collectDiagnostics = resourceApi.collectDiagnosticDataCommand(convertToCollectDiagnosticDataArguments(parameters));
+            PollingResult pollingResult = clouderaManagerPollingServiceProvider.startPollingCollectDiagnostics(stack, client, collectDiagnostics.getId());
```

Comment:
a log would be nice here. "starting polling blabla..." or something

Reply:
actually there is a log in startPollingCollectDiagnostics call itself